pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -207,6 +207,8 @@ checkout_ref(){
 
 # Update submodules
 init_submodules(){
+    [[ -z "$deepClone" ]] && [[ -z "$leaveDotGit" ]] && \
+    clean_git submodule update --init --recursive -j ${NIX_BUILD_CORES:-1} --depth 1 || \
```

Comment:
```suggestion
    # leaveDotGit exception is because shallow cloning will change hashes
    [[ -z "$deepClone" ]] && [[ -z "$leaveDotGit" ]] && \
    clean_git submodule update --init --recursive -j ${NIX_BUILD_CORES:-1} --depth 1 || \
```